### PR TITLE
Sb categoryitem/add background widget to category preset

### PR DESCRIPTION
**Pull request by Tim-M-S**
### All changes meet the following requirements
- [ ] Changelog entry was added
- [x] Changes have been tested by the author
- [ ] Changes have been tested by the reviewer
- [x] Plugin can be built

@plentymarkets/ceres-io

### CODE REVIEW

**Review by felixdausch — CHANGES_REQUESTED** [review]
Anscheinend gibt es merge-konflikte in der package.lock

File: src/Widgets/Common/BackgroundWidget.php [hunk 1]
```diff
@@ -31,6 +31,12 @@ public function getSettings()
 
         $settings->createCustomClass();
 
+        $settings->createSlider("opacity")
+            ->withDefaultValue(100)
+            ->withName("Widget.backgroundOpacityLabel")
+            ->withOption("inputInterval", 1)
+            ->withOption("inputMax", 100);
+
```
Comment (felixdausch) [inline src/Widgets/Common/BackgroundWidget.php]: Ist nochmal in Zeile 63++ definiert und m.M.n. dort auch sinnvoller positioniert

File: src/Widgets/Presets/DefaultItemCategoryPreset.php [hunk 2]
```diff
@@ -44,6 +55,94 @@ public function getWidgets()
...
+                                   ->withSetting("spacing.padding.top.unit", null)
+                                   ->withSetting("spacing.padding.bottom.value", 0)
+                                   ->withSetting("spacing.padding.bottom.unit", null)
+                                   ->withSetting("spacing.customMargin", true)
+                                   ->withSetting("spacing.margin.bottom.value", 0)
+                                   ->withSetting("spacing.margin.bottom.unit", null)
+                                   ->withSetting('text', $text);
+        }
+        
+    }
```
Comment (felixdausch) [inline src/Widgets/Presets/DefaultItemCategoryPreset.php]:
```
$codeWidget = $asChild
  ? $this->backgroundWidget->createChild('background', 'Ceres::CodeWidget')
  : $this->preset->createWidget('Ceres::CodeWidget');

$codeWidget->withSetting("customClass", "")
  ->withSetting("spacing.customPadding", true)
  ->withSetting("spacing.padding.left.value", 0)
  ->withSetting("spacing.padding.left.unit", null)
  ->withSetting("spacing.padding.right.value", 0)
  ->withSetting("spacing.padding.right.unit", null)
  ->withSetting("spacing.padding.top.value", 0)
  ->withSetting("spacing.padding.top.unit", null)
  ->withSetting("spacing.padding.bottom.value", 0)
  ->withSetting("spacing.padding.bottom.unit", null)
  ->withSetting("spacing.customMargin", true)
  ->withSetting("spacing.margin.bottom.value", 0)
  ->withSetting("spacing.margin.bottom.unit", null)
  ->withSetting('text', $text);
```

**Review by felixdausch — DISMISSED** [review]

File: src/Widgets/Common/BackgroundWidget.php (hunk 1 above)
Comment (Tim-M-S) [reply src/Widgets/Common/BackgroundWidget.php]: done

File: src/Widgets/Presets/DefaultItemCategoryPreset.php (hunk 2 above)
Comment (Tim-M-S) [reply src/Widgets/Presets/DefaultItemCategoryPreset.php]: done

**Review by Dominik2809 — APPROVED** [review]

**Pull request merged**